core/src/main/java/org/mule/construct/Flow.java

```diff
@@ -155,13 +137,40 @@ public MuleEvent process() throws Exception
     private MuleEvent createMuleEventForCurrentFlow(MuleEvent event, Object replyToDestination, ReplyToHandler
             replyToHandler)
     {
+        if (replyToHandler != null && !(replyToHandler instanceof DefaultReplyToHandler))
```

Comment:
This instance of seems ugly to me. Why only the DefaultReplyToHandler is a special case?